### PR TITLE
First attempt - Marius Larsen

### PR DESCRIPTION
**Pull request by mariuslarsen**
#### Check List:
- [X] Tests pass (`./test.sh mariuslarsen` shows no differences between expected and actual outputs)
- [X] All formatting changes by the build are committed
- [X] Your launch script is named `calculate_average_mariuslarsen.sh` (make sure to match casing of your GH user name) and is executable
- [X] Output matches that of `calculate_average_baseline.sh`
* Execution time: 8.944s on Intel® Core™ i7-10875H CPU @ 2.30GHz 8 cores
* Execution time of reference implementation: 4m4.129s

### CODE REVIEW

File: src/main/java/dev/morling/onebrc/CalculateAverage_mariuslarsen.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        int hashCode;
+        int negative;
+        int temperature;
+        byte current;
+        int offset;
+        while (buffer.hasRemaining()) {
+            hashCode = 1;
+            offset = buffer.position();
+            while ((current = buffer.get()) != DELIMITER) {
+                hashCode = 31 * hashCode + current;
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_mariuslarsen.java:86]: How would that deal with hash collisions?

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
I think this is prone to hash collisions?

**Comment by gunnarmorling** [comment]
I'm gonna close this one for now, as I haven't heard back in two days and need to clean the slate a bit. Feel free to re-open if you think the time has come. Thx!

File: src/main/java/dev/morling/onebrc/CalculateAverage_mariuslarsen.java (hunk 1 above)
Comment (mariuslarsen) [reply src/main/java/dev/morling/onebrc/CalculateAverage_mariuslarsen.java:86]: Badly 😅 
That may explain why the solution was pretty fast. I'll create a new PR when I find a better solution.

**Pull request closed without merging**